File: code/metodoOrdenamiento/quicksort/quickSort.py
def quickSort(list):
    if len(list) <= 1:
        return list
    pivot = list[len(list) // 2]
    izquierda = [x for x in list if x < pivot]
    medio = [x for x in list if x == pivot]
    derecha = [x for x in list if x > pivot]
    return quickSort(izquierda) + medio + quickSort(derecha)

print(quickSort([3,6,8,10,1,2,1]))

File: code/metodoOrdenamiento/quicksort/quickSort.java
package code.metodoOrdenamiento.quicksort;

public class quickSort {
    public static void QuickSort(int[] arr, int low, int high) {
        if (low < high) {
            int pi = partition(arr, low, high);
            QuickSort(arr, low, pi - 1);
            QuickSort(arr, pi + 1, high);
        }
    }
    private static int partition(int[] arr, int low, int high) {
        int pivot = arr[high];
        int i = low - 1;
        for (int j = low; j < high; j++) {
            if (arr[j] < pivot) {
                i++;
                int aux = arr[i];
                arr[i] = arr[j];
                arr[j] = aux;
            }
        }
        int aux = arr[i + 1];
        arr[i + 1] = arr[high];
        arr[high] = aux;
        return i + 1;
    }

    public static void main(String[] args) {
        int[] arr = { 10, 7, 8, 9, 1, 5 };
        int n = arr.length;
        QuickSort(arr, 0, n - 1);
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
